Tools/peg_generator/pegen/c_generator.py

```diff
@@ -55,10 +61,7 @@ def keyword_helper(self, keyword: str) -> Tuple[str, str]:
 
     def visit_NameLeaf(self, node: NameLeaf) -> Tuple[str, str]:
         name = node.value
-        if name in ("NAME", "NUMBER", "STRING"):
-            name = name.lower()
-            return f"{name}_var", f"_PyPegen_{name}_token(p)"
-        if name in ("NEWLINE", "DEDENT", "INDENT", "ENDMARKER", "ASYNC", "AWAIT"):
+        if name in self.non_exact_tokens:
             name = name.lower()
             return f"{name}_var", f"_PyPegen_{name}_token(p)"
```

Comment:
Hmmm, I think I still like to have the abstraction in case we need to intercede something like with the `NAME` token in the future (this also allows us to not have "hardcoded" names in the generator) but if you think is better to re-add that in the future and eliminate the wrappers I can totally do it as I don't feel strongly about it.

What do you think?

Reply:
> I think I still like to have the abstraction in case we need to intercede something like with the NAME token in the future

It feels, that that's only the case with `STRING`s. I don't really expect to do anything more complicated that just returning the token with all the whitespace ones (`INDENT`, `DEDENT` etc.) and I'd re-add those functions for `async` and `await`, if it were ever needed.

With that said, I really don't feel strongly about either one, so it's your call! 😃

Reply:
Ok, I am going to merge this and we can revisit this in further refactors of this code. I have some improvements in mind to avoid checking for sub-strings (like `var.startswith("name")` when distinguishing the types and we can explore doing this in that PR.

Reply:
I experimented with your suggestion and the main blocker is that `_PyPegen_lookahead` is called with these functions and it imposes that they may take the parser as the only argument and there is not a simple way to make `_PyPegen_lookahead` allow to forward the arguments :(

Reply:
I didn't think of this problem. Thanks a lot for doing the investigation!